pkg/agent/run.go

```diff
@@ -199,16 +203,28 @@ func gatherData(ctx context.Context, config Config) []*api.DataReading {
 	completedDataGatherers := make(map[string]bool, len(dataGatherers))
 
 	// Fetch from all datagatherers
-	getReadings := func() error {
+	getReadings := func() *multierror.Error {
 		var dgError *multierror.Error
 		for k, dg := range dataGatherers {
 			if completedDataGatherers[k] {
 				continue
 			}
 			dgData, err := dg.Fetch()
 			if err != nil {
-				err = fmt.Errorf("%s: %v", k, err)
-				dgError = multierror.Append(dgError, err)
+				if _, ok := err.(*k8s.CRDNotFoundError); ok {
```

Comment:
I think `_` here is the CRD error you want. e.g. this bit hopefully isn't needed.

```
err := k8s.CRDNotFoundError{
  Err:          err.Error(),
  DataGatherer: k,
}
```

Reply:
The purpose of this is to add the DataGatherer into the error. This could also be handled by giving `k` as an input to `dg.Fetch()` 
i.e. `func (g *DataGathererDynamic) Fetch(Datagatherer string) (interface{}, error) {`

Or perhaps adding the name of the data gatherer to the `DataGathererDynamic` structure is also a valid way to go about it